### PR TITLE
Book bloc logic

### PR DESCRIPTION
**Pull request by UrosLITS**
Added simple logic for adding bookPage into local list and uploading image of that page to the storage.

### CODE REVIEW

File: lib/presentation/bookPages/bloc/book_bloc.dart [hunk 1]
```diff
@@ -36,4 +45,53 @@ class BookBloc extends Bloc<BookEvents, BookState> {
...
+  }
+
+  Future<void> _onAddImageToServer(
+      AddImageToServerEvent event, Emitter<BookState> emit) async {
+    emit(LoadingBookPageState());
+
+    ui.Image? decodedImage;
+    File image = File(event.imageFile.path);
+
+    decodedImage = await decodeImageFromList(image.readAsBytesSync());
```
Comment (petarLits) [inline lib/presentation/bookPages/bloc/book_bloc.dart]: what if this call fail?

File: lib/presentation/bookPages/bloc/book_bloc.dart [hunk 2]
```diff
@@ -36,4 +45,53 @@ class BookBloc extends Bloc<BookEvents, BookState> {
...
+    BookPageImage bookPageImage = BookPageImage(
+      width: decodedImage.width,
+      height: decodedImage.height,
+      filePath: event.imageFile.path,
+    );
+    event.bookPage.bookPageImage = bookPageImage;
+
+    final result = await FirebaseDbManager.instance
```
Comment (petarLits) [inline lib/presentation/bookPages/bloc/book_bloc.dart]: what if this call fail?

**Review by petarLits — CHANGES_REQUESTED** [review]

File: lib/presentation/bookPages/bloc/book_events.dart [hunk 3]
```diff
@@ -25,3 +27,49 @@ class InitBookEvent extends BookEvents {
...
+    required this.bookID,
+    required this.bookPage,
+  });
+
+  final File imageFile;
+  final String bookID;
+  final BookPage bookPage;
+
+  @override
+  List<Object?> get props => [];
```
Comment (milicaLits) [inline lib/presentation/bookPages/bloc/book_events.dart]: stavi props

File: lib/presentation/bookPages/view/add_edit_new_page.dart [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            ),
+          ),
+        ),
+        floatingActionButton: Padding(
+          padding: const EdgeInsets.only(bottom: 60),
+          child: FloatingActionButton(
+            child: Icon(CupertinoIcons.camera),
+            onPressed: () async {
+              final XFile? result =
+                  await _imagePicker.pickImage(source: ImageSource.gallery);
```
Comment (milicaLits) [inline lib/presentation/bookPages/view/add_edit_new_page.dart:187]: you could add an event for picking image

File: lib/presentation/bookPages/view/add_edit_new_page.dart [hunk 5]
```diff
@@ -0,0 +1,201 @@
+import 'dart:io';
+
+import 'package:book/models/book/book_imports.dart';
+import 'package:book/presentation/bookPages/bloc/book_bloc.dart';
```
Comment (milicaLits) [inline lib/presentation/bookPages/view/add_edit_new_page.dart:3]: use part and part of for book bloc, event and state files

**Review by milicaLits — CHANGES_REQUESTED** [review]

**Review by milicaLits — APPROVED** [review]

File: lib/presentation/bookPages/bloc/book_bloc.dart [hunk 6]
```diff
@@ -68,23 +70,28 @@ class BookBloc extends Bloc<BookEvents, BookState> {
...
+
+      final result = await FirebaseDbManager.instance
+          .addImageToServer(event.bookPage, event.imageFile, event.bookID);
+
+      emit(UploadedImageToServerState(
+          isUploaded: result, bookPage: event.bookPage));
+      emit(LoadedBookPageState());
+    } on Exception catch (e) {
+      emit(ErrorState(
+          bookData: book.bookData!, error: e, pageIndex: currentPageIndex));
```
Comment (petarLits) [inline lib/presentation/bookPages/bloc/book_bloc.dart:95]: where is Loaded state?

**Review by petarLits — CHANGES_REQUESTED** [review]

**Review by petarLits — APPROVED** [review]

**Pull request merged**